Meadowlark-js/packages/meadowlark-core/src/validation/DocumentIdentityExtractor.ts

```diff
@@ -32,8 +32,18 @@ type NullableTopLevelEntity = { superclass: TopLevelEntity | null };
  */
 function singleIdentityFrom(property: EntityProperty, body: object, documentPath: string[]): DocumentIdentity {
   const { apiMapping } = property.data.meadowlark as EntityPropertyMeadowlarkData;
-  const documentPathAsString: string = [...documentPath, apiMapping.fullName].join('.');
-  const elementValue: string | undefined = R.path([...documentPath, apiMapping.fullName], body);
+
+  let path: string[];
+
+  if (property.type === 'schoolYearEnumeration' && property.roleName) {
```

Comment:
Looking at this again, I should have used the same string "null or empty" check that I employed on line 76 below.

Reply:
done